sympy/functions/special/beta_functions.py

```diff
@@ -118,3 +128,201 @@ def _eval_conjugate(self):
+        return Integral(t**(a - 1)*(1 - t)**(b - 1), (t, x1, x2))
+
+    def _eval_rewrite_as_hyper(self, a, b, x1, x2, **kwargs):
+        from sympy.functions.special.hyper import hyper
+        return (x2**a * hyper((a, 1 - b), (a + 1,), x2) - x1**a * hyper((a, 1 - b), (a + 1,), x1)) / a
+
+###############################################################################
+#################### REGULARIZED INCOMPLETE BETA FUNCTION #####################
+###############################################################################
+
+class regularized_betainc(Function):
+    r"""
+    The Generalized Regularized Incomplete Beta function is given by
+
+    .. math::
+        \mathrm{I}_{(x_1, x_2)}(a, b) = \frac{\mathrm{B}_{(x_1, x_2)}(a, b)}{\mathrm{B}(a, b)}
+
+    The Regularized Incomplete Beta function is a special case
+    of the Generalized Regularized Incomplete Beta function :
+
+    .. math:: \mathrm{I}_z (a, b) = \mathrm{I}_{(0, z)}(a, b)
+
+    The Regularized Incomplete Beta function is the cumulative distribution
+    function of the beta distribution.
+
+    Examples
+    ========
+
+    >>> from sympy import regularized_betainc, symbols, conjugate
+    >>> a, b, x, x1, x2 = symbols('a b x x1 x2')
+
+    The Generalized Regularized Incomplete Beta
+    function is given by:
+
+    >>> regularized_betainc(a, b, x1, x2)
+    regularized_betainc(a, b, x1, x2, 1)
+
+    The Regularized Incomplete Beta function
+    can be obtained as follows:
+
+    >>> regularized_betainc(a, b, 0, x)
+    regularized_betainc(a, b, 0, x, 1)
+
+    The Regularized Incomplete Beta function
+    obeys the mirror symmetry:
+
+    >>> conjugate(regularized_betainc(a, b, x1, x2))
+    regularized_betainc(conjugate(a), conjugate(b), conjugate(x1), conjugate(x2), 1)
+
+    We can numerically evaluate the Regularized Incomplete Beta
+    function to arbitrary precision on the whole complex plane:
+
+    >>> from sympy import regularized_betainc, pi, E
+    >>> regularized_betainc(1, 2, 0, 0.25).evalf(10)
+    0.4375000000
+    >>> regularized_betainc(pi, E, 0, 1).evalf(5)
+    1.00000
+
+    The Generalized Regularized Incomplete Beta function can be
+    expressed in terms of the Generalized Hypergeometric function.
+
+    >>> from sympy import hyper
+    >>> regularized_betainc(a, b, x1, x2).rewrite(hyper)
+    (-x1**a*hyper((a, 1 - b), (a + 1,), x1) + x2**a*hyper((a, 1 - b), (a + 1,), x2))/(a*beta(a, b))
+
+    See Also
+    ========
+
+    beta: Beta function
+    hyper: Generalized Hypergeometric function
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Beta_function#Incomplete_beta_function
+    .. [2] https://dlmf.nist.gov/8.17
+    .. [3] https://functions.wolfram.com/GammaBetaErf/Beta4/
+    .. [4] https://functions.wolfram.com/GammaBetaErf/BetaRegularized4/02/
+
+    """
+    nargs = 5
+    unbranched = True
+
+    def __new__(cls, a, b, x1, x2, reg=1):
+        return Function.__new__(cls, a, b, x1, x2, 1)
```

Comment:
What is `reg` for?

Reply:
`reg` is a boolean for regularized. This is the argument we have to pass in to mpmath while evaluating numerically. Is there a way to hide arguments from the user? Like a private argument so to say? If so, we could eliminate it from here

Reply:
Is `reg` always just `1`? If so then it shouldn't be in the args. Another way should be found to pass `reg=1` to mpmath.

Reply:
There is one way, where we define a fallback evaluation function `_imp_` for these classes. See this - 
https://github.com/sympy/sympy/blob/9e33ca3db9137ab00d989992f3f28db4c3db37e0/sympy/core/function.py#L551-L555
Inside `_imp_`, we would have to import `betainc` of mpmath, check if all arguments are non-symbolic and return the answer given by mpmath. For this to work, we should also make sure `betainc` does not match with mpmath in `_eval_evalf`.

I'm not sure how good this is though.

Reply:
I don't think that the `_imp_` function is suitable because it skips all of the logic below. It's possible that we need to add something in `Function._eval_evalf` to check for a new special method.

Reply:
If we're going to do this, it would be easy to remove `reg=1` from args.

Reply:
I have added `_eval_mpmath` to handle this case